### PR TITLE
Change AdamW from TorchANI to PyTorch

### PR DESCRIPTION
**Pull request by RMeli**
The AdamW optimizer seems to be supported by PyTorch as of version 1.2.

### CODE REVIEW

**Comment by RMeli** [comment]
@farhadrgh feel free to close the PR if this is the case. I didn't ask/open an issue since it was only a few small changes.

File: torchani/optim.py [hunk 1]
```diff
@@ -1,116 +0,0 @@
...
-            loss = closure()
-
-        for group in self.param_groups:
-            for p in group['params']:
-                if p.grad is None:
-                    continue
-
-                # Perform stepweight decay
-                # p.data.mul_(1 - group['lr'] * group['weight_decay'])  # AdamW
-                p.data.mul_(1 - group['weight_decay'])  # Neurochem
```
Comment (farhadrgh) [inline torchani/optim.py:71]: Just for the record, the NueroChem implementation applies the weight decay slightly different than the suggested algorithm in the [paper](https://arxiv.org/pdf/1711.05101.pdf). We are, however, keeping the weight decays in the example training files as they are and leave the choice of these hyperparameters to users for now.
